gcp_pilot/drive.py

```diff
@@ -0,0 +1,111 @@
+# Reference: https://developers.google.com/drive/api/reference/rest/v3
+
+from typing import Any
+
+from gcp_pilot.base import DiscoveryMixin, GoogleCloudPilotAPI
+
+
+class Drive(GoogleCloudPilotAPI):
+    _scopes = ["https://www.googleapis.com/auth/drive"]
+
+    def __init__(self, **kwargs):
+        super().__init__(serviceName="drive", version="v3", **kwargs)
+
+
+class Permissions(DiscoveryMixin, Drive):
+    @property
+    def get_service(self):
+        return self.client.permissions()
+
+    def create_for_domain(
+        self,
+        file_id: str,
+        role: str,
+        domain: str,
+        extra_permission_body: dict[str, Any] | None = None,
+        **kwargs,
+    ):
+        body = {
+            "role": role,
+            "type": "domain",
+            "domain": domain,
+        } | (extra_permission_body or {})
+
+        return self._execute(self.get_service.create, fileId=file_id, body=body, **kwargs)
+
+    def create_for_group(
+        self,
+        file_id: str,
+        role: str,
+        email: str,
+        extra_permission_body: dict | None = None,
+        **kwargs,
+    ):
+        body = {
+            "role": role,
+            "type": "group",
+            "emailAddress": email,
+        } | (extra_permission_body or {})
+        return self._execute(self.get_service.create, fileId=file_id, body=body, **kwargs)
+
+    def create_for_anyone(
+        self,
+        file_id: str,
+        role: str,
+        extra_permission_body: dict | None = None,
+        **kwargs,
+    ):
+        body = {
+            "role": role,
+            "type": "anyone",
+        } | (extra_permission_body or {})
+        return self._execute(self.get_service.create, fileId=file_id, body=body, **kwargs)
+
+    def create_for_user(
+        self,
+        file_id: str,
+        role: str,
+        email: str,
+        extra_permission_body: dict | None = None,
+        **kwargs,
+    ):
+        body = {
+            "role": role,
+            "type": "user",
+            "emailAddress": email,
+        } | (extra_permission_body or {})
+        return self._execute(self.get_service.create, fileId=file_id, body=body, **kwargs)
+
+    def delete(self, permission_id: str, file_id: str):
+        return self._execute(self.get_service.delete, fileId=file_id, permissionId=permission_id)
+
+    def list(self, file_id: str, **kwargs):
+        return self._execute(self.get_service.list, fileId=file_id, **kwargs)
```

Comment:
Isn't it paginated? We have a helper for those.

Reply:
good call..

I'm working on it